File: wakakusa/demo/points.rb
# A small class of values, carried on the app's own state.
require "wakakusa"

class Point
  attr_reader :x, :y

  def initialize(x, y = 0)
    @x = x
    @y = y
  end
end

class Points
  def initialize
    @sel = Point.new(3, 4)
    @dist = 0
  end

  def view
    column(
      text("p=(#{@sel.x}, #{@sel.y}) d2=#{@dist}"),
      row(
        button("right") { @sel = Point.new(@sel.x + 5, @sel.y) },
        button("swap") { @sel = Point.new(@sel.y, @sel.x) },
        button("measure") { @dist = @sel.x * @sel.x + @sel.y * @sel.y },
        spacing: 6.0
      ),
      spacing: 8.0,
      padding: 12.0
    )
  end
end

run(Points.new, title: "points")

File: gomamochi/demo/points.go
// A small struct of values, carried on the app's own state.
package main

import (
	"fmt"

	. "github.com/i2y/yokan/gomamochi"
)

type Point struct {
	x, y int
}

type Points struct {
	sel  Point
	dist int
}

func (p *Points) View() Element {
	return Column(
		Text(fmt.Sprintf("p=(%d, %d) d2=%d", p.sel.x, p.sel.y, p.dist)),
		Row(
			Button("right").OnClick(func() { p.sel = Point{p.sel.x + 5, p.sel.y} }),
			Button("swap").OnClick(func() { p.sel = Point{p.sel.y, p.sel.x} }),
			Button("measure").OnClick(func() { p.dist = p.sel.x*p.sel.x + p.sel.y*p.sel.y }),
		).Spacing(6),
	).Spacing(8).Padding(12)
}

func main() {
	Run(&Points{sel: Point{3, 4}}, Title("points"))
}
